pkg/controller/v1beta1/inferenceservice/reconcilers/cabundlesecret/cabundlesecret_reconciler.go

```diff
@@ -0,0 +1,162 @@
+		return fmt.Errorf("fails to reconcile ca bundle secret: %w", err)
+	}
+
+	return nil
+}
+
+func (c *CaBundleSecretReconciler) getCabundleSecretForUserNS(caBundleNameInConfig string, kserveNamespace string, isvcNamespace string) (*corev1.Secret, error) {
+	var newCaBundleSecret *corev1.Secret
+	log.Info("In caBundleNameInConfig")
+
+	// Check if cabundle Secret exist & the cabundle.crt exist in the data in controller namespace
+	// If it does not exist, return error
+	caBundleSecret := &corev1.Secret{}
+	if err := c.client.Get(context.TODO(),
+		types.NamespacedName{Name: caBundleNameInConfig, Namespace: kserveNamespace}, caBundleSecret); err == nil {
+
+		if cabundleSecretData := caBundleSecret.Data[constants.DefaultCaBundleFileName]; cabundleSecretData == nil {
+			return nil, fmt.Errorf("specified cabundle file %s not found in cabundle secret %s",
+				constants.DefaultCaBundleFileName, caBundleNameInConfig)
+		} else {
+			secretData := map[string][]byte{
+				constants.DefaultCaBundleFileName: cabundleSecretData,
+			}
+			newCaBundleSecret = getDesiredCaBundleSecretForUserNS(constants.DefaultGlobalCaBundleSecretName, isvcNamespace, secretData)
+		}
+	} else {
+		return nil, fmt.Errorf("can't read cabundle secret %s: %w", constants.DefaultCaBundleFileName, err)
+	}
+
+	return newCaBundleSecret, nil
+}
+
+func getDesiredCaBundleSecretForUserNS(secretName string, namespace string, cabundleData map[string][]byte) *corev1.Secret {
+	desiredSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: cabundleData,
+	}
+
+	return desiredSecret
+}
+
+// ReconcileCaBundleSecret will manage the creation, update and deletion of the ca bundle Secret
+func (c *CaBundleSecretReconciler) ReconcileCaBundleSecret(desiredSecret *corev1.Secret) error {
+	log.Info("Reconciling ReconcileCaBundleSecret")
+
+	// Create secret if does not exist
+	existingSecret := &corev1.Secret{}
+	err := c.client.Get(context.TODO(), types.NamespacedName{Name: desiredSecret.Name, Namespace: desiredSecret.Namespace}, existingSecret)
+	if err != nil {
+		if apierr.IsNotFound(err) {
+			log.Info("Creating cabundle secret", "namespace", desiredSecret.Namespace, "name", desiredSecret.Name)
+			err = c.client.Create(context.TODO(), desiredSecret)
+		}
+		return err
+	}
+
+	// Return if no differences to reconcile.
+	if equality.Semantic.DeepEqual(desiredSecret, existingSecret) {
+		return nil
+	}
+
+	// Reconcile differences and update
+	diff, err := kmp.SafeDiff(desiredSecret.Data, existingSecret.Data)
+	if err != nil {
+		return fmt.Errorf("failed to diff cabundle secret: %w", err)
+	}
+	log.Info("Reconciling cabundle secret diff (-desired, +observed):", "diff", diff)
+	log.Info("Updating cabundle secret", "namespace", existingSecret.Namespace, "name", existingSecret.Name)
+	existingSecret.Data = desiredSecret.Data
+	err = c.client.Update(context.TODO(), existingSecret)
```

Comment:
I think that if the secret is not owned by the controller, the update should be skipped. We don't want to overwrite something that wasn't created by the controller.

Reply:
the secret is created by kserve controller.

Reply:
If it does not exists, yes.
But are we OK on overwriting if there is an already existent secret that the user may have created if, by chance, has the same name?

Reply:
If users use kserve, then I think they should not use the secret name. like storage-config secret.
```
        "storageSpecSecretName": "storage-config"
```